src/Analyzers/CSharp/CodeFixes/UseCollectionInitializer/CSharpUseCollectionInitializerCodeFixProvider_CollectionInitializer.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.CSharp.UseObjectInitializer;
+using Microsoft.CodeAnalysis.Formatting;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.UseCollectionInitializer;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseCollectionInitializer
+{
+    using static SyntaxFactory;
+
+    internal partial class CSharpUseCollectionInitializerCodeFixProvider
+    {
+        private static BaseObjectCreationExpressionSyntax CreateObjectInitializerExpression(
+            BaseObjectCreationExpressionSyntax objectCreation,
+            ImmutableArray<Match<StatementSyntax>> matches)
+        {
+            var expressions = CreateCollectionInitializerExpressions();
+            var withLineBreaks = AddLineBreaks(expressions);
```

Comment:
the collection initializer case is pretty simple.  We create a list of all the expressions, then add lines breaks between the elements (we always do initializers across multiple lines).

Note: in the future we may want to copy the code that properly indents expressions so that when we move expressions into the collection initializer we do a good job with multi-line exprs.